### ISSUE TITLE
Using Fastlane actions as build phase scripts, bad idea?

### ISSUE THREAD

**Issue by fcy** [fastlane/fastlane#882]
Hello all,

I'm wondering if using Fastlane actions in build phase scripts is bad idea, meaning it is fragile and I should use something else. This is roughly what I have in mind:

```ruby
platform :ios do
#all my test and build stuff for the app
end

platform :xcode do
  lane :pre_compile do
    organize_imports
    clang_format
    overlay_icons
  end
end
```

In a Xcode Build Phase script I'd call `fastlane xcode pre_compile`.

What do you think? Should I use Make or Rake instead? I'd rather use one tool, but I also don't want to use it "wrongly".

**Comment by KrauseFx** [comment]
Interesting idea, I can't think of any reason why this wouldn't be a good idea, so go ahead and try it :+1: 

Let me know how it goes

**Comment by fcy** [comment]
First problem, can't use platform `xcode`: "RuntimeError: Platform 'xcode' is not supported. Must be either [:ios, :mac, :android]". This is the sentry id `09cf99878fa2c424e6d578ba5a98f07b`.

**Comment by KrauseFx** [comment]
```ruby
platform :xcode do
  lane :pre_compile do
    organize_imports
    clang_format
    overlay_icons
  end
end
```

Platform 'xcode' is not supported. You must use either `:ios`, `:mac` or `:android`

**Comment by fcy** [comment]
I've created a `xcode_pre_compile` lane instead of having it in a `xcode` platform.

More on this, implementing the actions was easy. I just created them inside `actions`. I'm using Bundler to make sure everyone has the same version of all gems so I need to `bundle exec fastlane` but Xcode's Build Phase Script can't find `bundle`, I've tried all the tricks I know including the `#!/bin/bash --login`. I'll keep trying  but if anyone has any ideas I'm all ears.
